### ISSUE TITLE
Feature Request: Implement additional checks on ParamSpec Components

### ISSUE THREAD

**Issue by MHDante** [microsoft/pyright#1601]
The file below represents examples from PEP 612 (https://www.python.org/dev/peps/pep-0612/#id2)

This FR recommends checks for the following issues surrounding ParamSpec
 - Allow single use of Paramspec variable (perhaps should be own issue)
 - Raise error when P.kwargs is used for *args
 - Raise error when P.args is used for **kwargs
 - Raise error when ParamSpec components used as type hints for local vars
 - Raise error when P.args is used on its own
 - Raise error when P.kwargs is used on its own
 - Raise error when ParamSpec Components are used outside of scope
 
 The following file tests the cases listed above:
 
 ```python
 from typing import Callable, TypeVar, ParamSpec

P = ParamSpec("P")
R = TypeVar("R")

def puts_p_into_scope(f: Callable[P, int]) -> None:          # Should allow single use.

  def inner(*args: P.args, **kwargs: P.kwargs) -> None:      # This is fine (currently accepts)
    pass

  def mixed_up(*args: P.kwargs, **kwargs: P.args) -> None:   # Should raise error
    pass

  def misplaced(x: P.args) -> None:                          # Should raise error
    pass

  stored_args: P.args                                        # Should raise error
  stored_kwargs: P.kwargs                                    # Should raise error

  stored_args = "bad string"                                 # Should raise error. 
  stored_kwargs = "bad string"                               # Should raise error. 

  def just_args(*args: P.args) -> None:                      # Should raise error
    pass

  def just_kwargs(**kwargs: P.kwargs) -> None:               # Should raise error
    pass

  #Used to quiet reportUnusedVariable
  print(inner, mixed_up, misplaced, stored_args, 
        stored_kwargs, just_args, just_kwargs)

def out_of_scope(*args: P.args, **kwargs: P.kwargs) -> None: # Should raise error
  pass

out_of_scope_var1 : R = 12                                   # this is fine (currently raises error)

out_of_scope_var2 : P = 12                                   # this is fine (currently raises error)

out_of_scope_var3 : P.args = 12                              # should raise error
```

Paramspec is not out yet, so I don't think this would be a high impact feature.  But I thought I would document it anyway.

My main use for this feature is to help with avoiding errors while writing decorators, but it's not blocking me in any way

**Comment by erictraut** [comment]
Thanks for the bug report. I've addressed everything in the sample above and included this as a unit test.

This is included in Pyright 1.1.119, which I just published. It will also be included in the next version of Pylance.
